service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ObservationStatementMapperTest.java

```diff
@@ -173,6 +177,16 @@ public void When_MappingParsedObservationJsonWithNoDates_Expect_Exception() thro
             -> observationStatementMapper.mapObservationToObservationStatement(parsedObservation, true));
     }
 
+    @Test
+    public void When_MappingParsedObservationJsonWithUnmappedPerformer_Expect_Exception() throws IOException {
+        var jsonInput = ResourceTestFileUtils.getFileContent(INPUT_JSON_WITH_PARTICIPANT);
+        Observation parsedObservation = new FhirParseService().parseResource(jsonInput, Observation.class);
+
+        assertThatThrownBy(() -> observationStatementMapper.mapObservationToObservationStatement(parsedObservation, false))
+            .isExactlyInstanceOf(EhrMapperException.class)
+            .hasMessageStartingWith("No ID mapping for reference Practitioner/");
```

Comment:
Why not assert the full message, since we're setting the reference?

Reply:
Just because the actual reference isn't visible in the test, but since that kind of thing is the norm in this codebase I don't mind making this change.